### ISSUE TITLE
hackage-packages.nix: systemd and spacecookie use wrong versions of dependencies

### ISSUE THREAD

**Issue by sternenseemann** [NixOS/nixpkgs#75119]
`spacecookie` is currently marked as broken, since in its dependencies `systemd` is specified instead of `systemd_2_2_0` which would be required (constraint `>= 2.1.0`). `systemd_2_2_0` in turn is also broken, since it uses `network` instead of `network_3_1_1_1` which would be needed.

From looking at `hackage2nix` I couldn't figure out, what exactly goes wrong when figuring out the dependencies and their constraints for those two packages.

Any pointers helping me to fix this would be appreciated!

cc @peti

**Comment by cdepillabout** [comment]
@sternenseemann The haskell stuff in nixpkgs follows the most recent Stackage LTS versions by default:

https://github.com/NixOS/nixpkgs/blob/c7db5662788a1f6f294c6a07980853ffb66a0154/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L45-L46

It looks like `systemd` is in the above LTS, which locks its version at `1.2.0`:

https://github.com/NixOS/nixpkgs/blob/c7db5662788a1f6f294c6a07980853ffb66a0154/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L2052

You can build this `systemd` derivation like the following:

```console
$ cd nixpkgs/
$ nix-build -A haskellPackages.systemd
```

However, `hacakge2nix` also generates a derivation for the latest release of `systemd` on Hackage:

```console
$ nix-build -A haskellPackages.systemd_2_2_0
```

You'll need to create an override for `spacecookie` that uses the latest version of `systemd`.  Here's an example:

https://github.com/NixOS/nixpkgs/blob/c7db5662788a1f6f294c6a07980853ffb66a0154/pkgs/development/haskell-modules/configuration-common.nix#L1123-L1126

However, like you say, it looks like `systemd_2_2_0` requires a later version of `network`, so you'll have to create that override as well.

----------------------------------------------------------

Please send a PR adding the required overrides to `configuration-common.nix`, and marking `spacecookie` as not broken (by following this video: https://discourse.nixos.org/t/video-tutorial-how-to-fix-broken-haskell-packages-in-nix/3968).

**Comment by sternenseemann** [comment]
Thanks, will do!

**Comment by cdepillabout** [comment]
Closed in #75147
